task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,266 @@
+                .;  # The accumulator is the image object itself
+
+                # Check if a valid mirror exists for the given source
+                if ($safe_mapping[$source] | type) == "array" and ($safe_mapping[$source] | length > 0) then
+                  $safe_mapping[$source][0] as $mirror |
+
+                  # Check if the image URL starts with the source registry
+                  if .image | startswith($source) then
+                    # KEY CHANGE: Verify what follows the match
+                    ($source | length) as $len |
+                    (.image | .[$len:$len+1]) as $next_char |
+
+                    # Perform replacement ONLY if the match is followed by the end of the string, ":", or "@"
+                    if ($next_char == "" or $next_char == ":" or $next_char == "@") then
+                      # Replace the prefix with the mirror and append the rest of the string
+                      .image |= ($mirror + (. | ltrimstr($source)))
+                    else
+                      . # This is a partial match (e.g., "operator-example"), so do nothing
+                    end
+                  else
+                    . # If it does not start with the source, do nothing
+                  end
+                else
+                  . # If no mirror exists for the source, do nothing
+                end
+              );
+
+            # Process the main JSON object
+            if has("relatedImages") and (.relatedImages | type == "array") then
+              .relatedImages |= map(replace_image_url)
+            else
+              .
+            end
+          ' < "$INPUT_CATALOG_FILE" > "$CATALOG_OUTPUT_TMP_FILE"
+
+          # --- Finalization ---
+          mv "$CATALOG_OUTPUT_TMP_FILE" "$CATALOG_FILE"
+          chmod a+r "$CATALOG_FILE"
+
+          echo "Update of relatedImages in ${CATALOG_FILE} has been completed."
+          # Temporary files are cleaned up automatically by the 'trap' command at the beginning of the script.
+        - "bash"
+        - "$(params.IDMS_PATH)"
+        - "$(params.OPM_OUTPUT_PATH)"
+      computeResources:
+        limits:
+          memory: 2Gi
+        requests:
+          cpu: 500m
+          memory: 2Gi
+    - name: validate-catalog
+      image: registry.redhat.io/openshift4/ose-operator-registry-rhel9:v4.18
+      workingDir: /var/workdir/source
+      env:
+        - name: OPM_OUTPUT_PATH_PARAM
+          value: $(params.OPM_OUTPUT_PATH)
+      securityContext:
+        runAsUser: 0
+      command: ["/bin/bash", "-c"]
+      args:
+        - |
+          set -euo pipefail
+          echo "Running OPM validate command"
+          # Get the directory part of the path
+          OUTPUT_DIR=$(dirname "${OPM_OUTPUT_PATH_PARAM}")
+
+          echo "Checking: ${OUTPUT_DIR}"
+          opm validate "${OUTPUT_DIR}"
+
+          echo "OPM validate finished"
+      computeResources:
+        requests:
+          memory: 3Gi
```

Comment:
Please set `requests` and `limits` to the same values.

https://home.robusta.dev/blog/kubernetes-memory-limit